### ISSUE TITLE
[Question] How is badges on Android set?

### ISSUE THREAD

**Issue by fungilation** [phonegap-build/PushPlugin#436]
I only see doc for iOS's side of the api setting for badges, none for Android. I'm not an Android user myself so this maybe a dumb question, but I see other's androids have common apps showing badge counts. How would badges be set on notification on android?

**Comment by artiveng** [comment]
I am facing same issue. I am looking for setting badge count in android. How do I do that? Any help would be appreciated.

**Comment by stevelacy** [comment]
@artiveng @fungilation  to set the 'badge' count with this plugin use: 
`msgcnt: Number`
Ref: https://github.com/phonegap-build/PushPlugin/blob/master/src/android/com/plugin/gcm/GCMIntentService.java#L120

**Comment by fungilation** [comment]
I see, thanks! This should definitely be noted in Readme.

**Comment by jdinse** [comment]
@stevelacy a few more sentences how to use it would be awesome. I still don't get it where I have to set `msgcnt`. Do I have to send it on server-side inside the payload to receive and evaluate with this plugin?

**Comment by stevelacy** [comment]
@jdinse Yes, the msgcnt is created and sent from the server to GCM server to the client.
A simple server example: https://github.com/stevelacy/phonegap-gcm/blob/master/send-gcm.coffee

And the JS alternative:

```js
var gcm = require('node-gcm');

var sender = new gcm.Sender('API KEY');

var devices = [];

var message = new gcm.Message({
  collapseKey: 'demo',
  timeToLive: 500,
  data: {
    title: 'phonegap gcm',
    message: 'GCM Message',
    msgcnt: 300
  }
});

devices.push('DEVICE ID');
sender.send(message, devices, 4, function(data) {
  return console.log(data);
});

```

**Comment by jdinse** [comment]
Thanks, man! I'm moving forward. I get the value of `msgcnt` in the notification popup when the app is in foreground, but there is still no badge upon the icon when the app is in background. So I wonder what I'm still missing. Shouldn't the badge appear upon the launch icon at the moment of the incoming push notification? However there is no badge at all.
I'm using PushPlugin 2.4.0 with Cordova 4.1.2 in an Intel XDK environment.

**Comment by kentmw** [comment]
@jdinse, it looks like setting the msgcnt sets the NotificationCompat.Builder.setNumber which has the following documentation:

>Set the large number at the right-hand side of the notification. This is equivalent to setContentInfo, although it might show the number in a different font size for readability.

It doesn't look like its badge number on the app icon but rather the number for the notification

**Comment by macdonst** [comment]
@kentmw +1

**Comment by ravisuryawanshi** [comment]
Hi,

I am getting badge count on the notification bar. I am not sure why it is coming. Can anyone help me?

Thanks
Ravi

Sample json

{
	"to": "fgl_nDuY8rI:APA91bEEqsWlQnrooiwD2dFfIRypFuG7lHOdE7OnJVTkS9-QnGffy6Evzu_1mSKYtMZvJQLn7S4VNLJ8LHpXn39R0qJPryQu7ljKOdBxPQxjUwOpB8EmfWd3Wx-hdqRMe09vxsyeSpl8",
	"data": {
		"notId": -451796093,
		"body": "You are invited to join room \"sdsdsd\" by Jack Wagner",
		"title": "New Invitation ",
		"badge": 20,
		"fromUser": {
			"display_name": "Jack Wagner",
			"userName": "jwagner"
		},
		"invitationTime": "3:14 PM IST Feb 9, 2017",
		"gcm": "You are invited to join room \"sdsdsd\"",
		"roomId": 1296,
		"channel": "ESG Room Notification Channel",
		"room": {
			"Name": "sdsdsd",
			"Owner": {
				"Name": "jwagner",
				"FirstName": "Jack",
				"Company": {
					"CompanyName": "CARRIER CO"
				},
				"DisplayName": "Jack Wagner",
				"LastName": "Wagner"
			},
			"Type": "Room",
			"ID": 1296,
			"SysId": 10296
		}
	}
}
![99edd142-ee3d-11e6-9fe3-b3878786538a](https://cloud.githubusercontent.com/assets/12716503/22777874/c09cc80a-eeda-11e6-9b48-37422387dae8.JPG)

**Comment by stephenlacy** [comment]
@ravisuryawanshi This issue is years old, I'd suggest opening a new one.

You only posted your GCM event data, not your application code which matters more for displaying the notification

And there is a newer fork: https://github.com/phonegap/phonegap-plugin-push
